cl/lib/search_utils.py

```diff
@@ -1199,3 +1201,43 @@ async def clean_up_recap_document_file(item: RECAPDocument) -> None:
         item.page_count = None
         item.is_available = False
         await item.asave()
+
+
+def store_search_query(request: HttpRequest, search_results: dict) -> None:
+    """Saves an user's search query in a SearchQuery model
+
+    :param request: the request object
+    :param search_results: the dict returned by `do_search` or
+        `do_es_search` functions
+    :return None
+    """
+    is_error = search_results.get("error")
+    is_es_search = search_results.get("results_details") is not None
+
+    search_query = SearchQuery(
+        user=None if request.user.is_anonymous else request.user,
+        get_params=request.GET.urlencode(),
+        failed=is_error,
+        query_time_ms=None,
+        hit_cache=False,
+        source=SearchQuery.WEBSITE,
+        engine=SearchQuery.ELASTICSEARCH if is_es_search else SearchQuery.SOLR,
+    )
+    if is_error:
+        # Leave `query_time_ms` as None if there is an error
+        search_query.save()
+        return
+
+    if is_es_search:
+        # We set ES `query_time` metadata with values 0 or 1 if cache is hit:
+        # 0: homepage cache; 1: micro cache
+        search_query.query_time_ms = ceil(search_results["results_details"][0])
+        search_query.hit_cache = search_query.query_time_ms in [0, 1]
```

Comment:
I was doing some testing and noticed that the queries used for stats on the homepage are not stored because they don't reach the `store_search_query` method.

Maybe it's fine not to store them, since these are queries we already know?

If that's the case, we could simplify the condition, add a comment, and remove 0 from the options here, making it clear that homepage queries are not stored.

Reply:
I think we shouldn't store the homepage query, since it's not exactly an user query but something we set ourselves and show the user as default. I simplified the condition as you recommended